Dev2

mkg20001 opened the pull request.

Commit:
Fixed some linter messages.
Refactored api render code to be more maintainable and expandable.
Removed the ability to query from stub-db since we use ids in mongo (so the tests would pass but the api would fail)

Commit:
Added a basic jsapi and started using it's get methods

Commit:
Some changes so we dont need to explicitly init the jsapi

mkg20001 commented on src/render/jsapi.js:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    const setRoute = modelName => {
+      return route(`set${C(modelName)}`, `(id, val, userId = 0) { // using user 0 since we are the server
```
not sure if using NULL/unset is better as this is somehow special from 0 which is also a number, but this could cause trouble with the SQL based schemas

BierLiebHaber replied:
Yeah I wasn't sure how to handle the internal api requests if we have logging enabled, for now I just wanted a way to distinguish http-api from js-api calls for debugging.
We should probably add an option so the user can decide what the internal id should be.

mkg20001 commented on src/render/hapi.js:
```diff
@@ -1,236 +1,225 @@
+  })`
     }
 
-    return L(`
```
nice cleanup

mkg20001 committed:
chore: upgrade deps

mkg20001 reviewed (commented):
The "delete new topic" test is failing, which previously wasn't the case

```
Debug: internal, implementation, error 
    ReferenceError: remove is not defined
    at handler (/tmp/celarium/0.7809832849311829/api.js:751:69)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async exports.Manager.execute (/home/maciej/Projekte/celarium/node_modules/@hapi/hapi/lib/toolkit.js:60:28)
    at async Object.internals.handler (/home/maciej/Projekte/celarium/node_modules/@hapi/hapi/lib/handler.js:46:20)
    at async exports.execute (/home/maciej/Projekte/celarium/node_modules/@hapi/hapi/lib/handler.js:31:20)
    at async Request._lifecycle (/home/maciej/Projekte/celarium/node_modules/@hapi/hapi/lib/request.js:370:32)
    at async Request._execute (/home/maciej/Projekte/celarium/node_modules/@hapi/hapi/lib/request.js:279:9)
```

mkg20001 commented:
I've pushed a commit to update all dependencies

mkg20001 commented:
I'll further review prob wednesday or thursday

If you can you should extend the tests, so you can iterate more confidently. Mostly copy-paste replacing manual browser API testing.

Should we consider adding tests part of the deal or do you want something extra for that?

BierLiebHaber commented:
Regarding tests: I consider them part of the individual milestones and I'll try to fix all existing test and add a few more for the js-api today.

BierLiebHaber committed:
fixed delete new topic test
added isSymbolic function stub (TODO: implement isSymbolic)
fixed some more linter messages

BierLiebHaber committed:
fixed the formatting of the generated code so its easier to read/debug

mkg20001 commented on src/render/jsapi.js:
```diff
@@ -0,0 +1,38 @@
+  const { validateAcls } = DBM
+  const accessLog = false
+  return {
+${S(routes).slice(0, -1) // slice to remove the last ","
```
trailing commas are a non-issue on js

mkg20001 commented on test/db.js:
```diff
@@ -1,15 +1,14 @@
 const {
-  generateTests,
-  generateCode,
+  // generateTests, // TODO: find use
```
this was copied from another project, it's basically a function that processes an array of [input, expectedOutput]

It will come in handy later when sync functions will be tested

mkg20001 reviewed (commented):
- formatting in generated output is a non-issue, we explicitly have an option for automatic formatting fixes by the linter
- jsapi should have an api more like what is generated in src/render/client.js (out/client.js)
- the hapi rewrite LGTM

I'll go ahead and merge those things that we have so far and push to master

I've created some issues to track the things that haven't been resolved

Wasn't quite keeping track of milestones done but I'm pretty sure that I owe you something now

mkg20001 commented:
Also heads up: I'll need this library in 2021 in production for some projects (the stable release for those would be something estimated arround q2 2021). Should probably be doable with the features available right now and a working ACL.

mkg20001 committed:
refactor: nitpicks
